streams/shared/src/main/scala/zio/stream/ZTransducer.scala

```diff
@@ -124,10 +124,41 @@ object ZTransducer {
     }
 
   /**
-   * Creates a transducer accumulating incoming values into lists of maximum size `n`.
+   * Creates a transducer accumulating incoming values into chunks of maximum size `n`.
    */
-  def collectAllN[I](n: Long): ZTransducer[Any, Nothing, I, List[I]] =
-    foldUntil[I, List[I]](Nil, n)((list, element) => element :: list).map(_.reverse).filter(_.nonEmpty)
+  def collectAllN[I](n: Int): ZTransducer[Any, Nothing, I, Chunk[I]] =
```

Comment:
Implementation detail that bubbled up due to the Chunk's size being `Int`, might be a deal breaker for this change?

Affects https://github.com/zio/zio/pull/3886/files#diff-e2c16046e05624c39e46b9e4c2870b9cR1596 and https://github.com/zio/zio/pull/3886/files#diff-e2c16046e05624c39e46b9e4c2870b9cR1603.

Reply:
This is a fair compromise, `n` is fine as an `Int` here.